drivers/isl/isl.cpp

```diff
@@ -0,0 +1,350 @@
+  write_reg(0x77, 0xB1); //Dist Th 2 Low LSB
+  write_reg(0x78, 0x0F); //Dist Th 3 Hgh MSB
+  write_reg(0x79, 0x5E); //Dist Th 3 Hgh LSB
+  write_reg(0x7A, 0x03); //Dist Th 3 Low MSB
+  write_reg(0x7B, 0xD7); //Dist Th 3 Low LSB
+  write_reg(0x7D, 0x95); //Mag Th 1 Hgh
+  write_reg(0x7E, 0x02); //Mag Th 1 EXP
+  write_reg(0x7F, 0xC7); //Mag Th 1 Low
+  write_reg(0x81, 0xA3); //Mag Th 2 Hgh
+  write_reg(0x82, 0x06); //Mag Th 2 EXP
+  write_reg(0x83, 0x94); //Mag Th 2 Low
+  write_reg(0x84, 0x02); //Motion Dist Th MSB
+  write_reg(0x85, 0x4E); //Motion Dist Th LSB
+  write_reg(0x86, 0x00); //Motion MAG Th
+  write_reg(0x90, 0x0F); //Driver Range
+  write_reg(0x91, 0x80); //Emitter DAC
+  write_reg(0x92, 0x00); //Driver Control
+  write_reg(0x93, 0x00); //Thresh DAC
+  write_reg(0x94, 0x00); //Driver Boost
+  write_reg(0x96, 0x00); //Driv Chg Bal DAC
+  write_reg(0x97, 0x00); //Front End Ctl
+  write_reg(0x98, 0x00); //AFE Ctl Regs
+  write_reg(0xA5, 0x00); //Emitter ADC Offset
+  write_reg(0xA7, 0x00); //Temp Sens Reg A
+  write_reg(0xA8, 0x00); //Temp Sens Reg B
+  write_reg(0xA9, 0x00); //Temp Sens Reg C
+  write_reg(0xAA, 0x00); //Temp Sens ADC Mode
+  write_reg(0xae, 0x00); //Internal Reset
+  write_reg(0x31, 0x00); //Temp Reference
+  write_reg(0x33, 0x00); //Phase Exponent
+  write_reg(0x34, 0x00); //Temp Coeff 1
+  write_reg(0x39, 0x00); //Temp Coeff 2
+  write_reg(0x36, 0x00); //Ambient Coeff 1
+  write_reg(0x3B, 0x00); //Ambient Coeff 2
+  unsigned char sample_delay_reg = read_reg(0x10); // Sample Len
+  _isl_sample_delay = calc_sample_delay(sample_delay_reg);
+  return 0;
+}
+
+//Calculate sample period as per value received from Sensor
+//calculations based on information provided by intersil
+int ISL::calc_sample_delay(unsigned char value)
+{
+  float d = 0.0711 * pow(2, value);
+  return (int) d + 1;
+}
+
+//Register write function call for I2C bus
+int ISL::write_reg(uint8_t address, uint8_t data)
+{
+	return _writeReg(address, &data, sizeof(data));
+}
+
+//Register read function call for I2C bus
+uint8_t ISL::read_reg(uint8_t address)
+{
+	uint8_t ret = 0;
+	if(_readReg(address, &ret, sizeof(ret)) == -1) {
+		_read_failure = true;
+		DF_LOG_ERR("Read Failure");
+	}
+	return ret;
+}
+
+//Stop ISL distance sensor driver
+int ISL::stop()
+{
+	int result = DevObj::stop();
+
+	if (result != 0) {
+		DF_LOG_ERR("DevObj stop failed");
+		return result;
+	}
+
+	usleep(100000);
+
+	result = devClose();
+
+	if (result != 0) {
+		DF_LOG_ERR("device close failed");
+		return result;
+	}
+
+	usleep(100000);
+	return 0;
+}
+
+//ISL sensor calibration routine:
+//User is required to put sensor at REF_DIST and run this routine to get
+//calibration value.
+//TODO: Make the calculated offset persistent using px4 param system
+int ISL::calibration(void)
+{
+	uint16_t raw_data;
+	float raw_avg = 0;
+	write_reg(0x2f, 0);	//DIST_CAL_OFF_MSB
+	write_reg(0x30, 0);	//DIST_CAL_OFF_LSB
+	usleep(100000);
+	for(uint8_t i = 0; i< 100; i++) {
```

Comment:
Don't use a non-native type for a loop variable ever. It will result in more CPU cycles and no saving in flash or ram. Use `unsigned` instead, which will give you the native unsigned type.